pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/LoadSheddingTask.java

```diff
@@ -40,19 +42,27 @@ public class LoadSheddingTask implements Runnable {
 
     private volatile ScheduledFuture<?> future;
 
+    private final ManagedLedgerFactory factory;
+
     public LoadSheddingTask(AtomicReference<LoadManager> loadManager,
                             ScheduledExecutorService loadManagerExecutor,
-                            ServiceConfiguration config) {
+                            ServiceConfiguration config,
+                            ManagedLedgerFactory factory) {
         this.loadManager = loadManager;
         this.loadManagerExecutor = loadManagerExecutor;
         this.config = config;
+        this.factory = factory;
     }
 
     @Override
     public void run() {
         if (isCancel) {
             return;
         }
+        if (factory instanceof ManagedLedgerFactoryImpl
```

Comment:
@horizonzy Could you review this behavior to use an implementation directly?

Reply:
We would better export ManagedLedgerFactory#getMetaStore, then check whether the metaStore is available or not.

Reply:
I think we only need to add a interface in ManagedLedgerFactory named isMetadataServiceAvailable() is enough, right? it seems we don't use any MetaStore interface now

Reply:
@horizonzy @poorbarcode because we don't have any other place use this method and ManagedLedgerFactory have only one implementation, so I suggest use instanceof is enough, if we have any other place use this method, change it to become interface is not too late.   ManagedLedgerFactory add a interface is a too heavy process.